runtime/src/main/java/org/corfudb/runtime/collections/ICorfuQueue.java

```diff
@@ -0,0 +1,94 @@
+package org.corfudb.runtime.collections;
+
+import org.corfudb.annotations.Accessor;
+import org.corfudb.annotations.ConflictParameter;
+import org.corfudb.annotations.MutatorAccessor;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Persisted Queue supported by CorfuDB using distributed State Machine Replication.
+ * Entries enqueued could be backed by a CorfuMap where each entry is mapped to a unique generated
+ * id that is returned upon successful <b>enqueue()</b>.
+ * Consumption is via the <b>entryList()</b> which returns entries along with their ids sorted by
+ * the order in which their <b>enqueue()</b> operations materialized. Thus a logical FIFO queue.
+ * Instead of a dequeue() this Queue supports a <b>remove()</b> which accepts the id of the element.
+ * Entries cannot be modified in-place and but can be removed from anywhere.
+ * Created by Sundar Sridharan on 5/8/19.
+ * @param E entry to enqueue into the persisted queue
+ */
+public interface ICorfuQueue<E> extends ISMRObject {
```

Comment:
please don't use `I` prefix

Reply:
done. (No comments on the abundance of I in interface definitions in existing code :P)

Reply:
Also I think I will get rid of the pure interface itself since there is only one class that is going to implement it. [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle.